### PR TITLE
Public Preview docs changes

### PR DESCRIPTION
**Pull request by lzdanski**
Public preview docs:

- Updates notice on https://docs.astronomer.io/astro/deploy-github-integration to `Public Preview`
- Adds release note for public preview to May 8, 2024 release notes
- Adds notes to relevant cross-linked docs about the public preview feature

### CODE REVIEW

File: astro/ci-cd-templates/github-actions-deploy-action.md [hunk 1]
```diff
@@ -8,6 +8,12 @@ description: Use pre-built Astronomer CI/CD templates to automate deploying Apac
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
+:::tip
+
+The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```
Comment (paolaperaza) [inline astro/ci-cd-templates/github-actions-deploy-action.md:13]: I'd add half a sentence on the git metadata you get. That's the real carrot!

File: astro/best-practices/connections-branch-deploys.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ To implement this use case:
...
 3. Choose a Deployment in your Workspace that you want to run your production workflows. In the Deployment, override the connections you configured to instead connect to your production resources. See [Override connection fields](create-and-link-connections.md#override-connection-fields).
-4. In the Git repository for your Astro project, define a multi-branch CI/CD pipeline for deploying to Astro. For an example of how to do this in GitHub, see the multibranch option in [CI/CD deploy action templates](/ci-cd-templates/github-actions-deploy-action.md#setup).
+4. In the Git repository for your Astro project, define a multi-branch CI/CD pipeline for deploying to Astro. For an example of how to do this in GitHub, see the multibranch option in [CI/CD deploy action templates](/ci-cd-templates/github-actions-deploy-action.md#setup) or with the [Astro Github integration](deploy-github-integration.md).
```
Comment (kentdanas) [inline astro/best-practices/connections-branch-deploys.md:41]:
```suggestion
4. In the Git repository for your Astro project, define a multi-branch CI/CD pipeline for deploying to Astro. For an example of how to do this in GitHub, see the [Astro Github integration](deploy-github-integration.md).
```
I would be in favor of just recommending the GH integration here, for simplicity. It's appropriate for this use case and more straightforward to implement.

File: astro/ci-cd-templates/github-actions-deploy-action.md (hunk 1 above)
Comment (lzdanski) [inline astro/ci-cd-templates/github-actions-deploy-action.md:13]:
```suggestion
The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action, allowing you to see Git metadata directly from your Astro Deployment. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```

File: astro/ci-cd-templates/github-actions-deployment-preview.md [hunk 3]
```diff
@@ -8,6 +8,12 @@ description: Use pre-built Astronomer CI/CD templates to automate deploying Apac
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
+:::tip
+
+The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```
Comment (lzdanski) [inline astro/ci-cd-templates/github-actions-deployment-preview.md:13]:
```suggestion
The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action, allowing you to see Git metadata directly from your Astro Deployment. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```

File: astro/ci-cd-templates/github-actions-private-network.md [hunk 4]
```diff
@@ -8,7 +8,13 @@ description: Use pre-built Astronomer CI/CD templates to automate deploying Apac
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-If you don't have access to the Astronomer [deploy action](https://github.com/astronomer/deploy-action) because you can't access the public internet from your GitHub repository, use one of the following private network templates to deploy to Astro. 
+:::tip
+
+The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```
Comment (lzdanski) [inline astro/ci-cd-templates/github-actions-private-network.md:13]:
```suggestion
The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action, allowing you to see Git metadata directly from your Astro Deployment. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```

File: astro/first-dag-github-actions.md [hunk 5]
```diff
@@ -4,7 +4,11 @@ title: 'Run your first DAG with GitHub Actions'
 id: 'first-dag-github-actions'
 ---
 
+:::tip
+
+The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```
Comment (lzdanski) [inline astro/first-dag-github-actions.md:9]:
```suggestion
```suggestion
The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action, allowing you to see Git metadata directly from your Astro Deployment. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```
```

**Review by TJaniF — APPROVED** [review]
LGTM :)

File: sidebarsAstro.js [hunk 6]
```diff
@@ -88,6 +88,7 @@ module.exports = {
         "deploy-project-image",
         "deploy-dags",
         "deploy-history",
+        "deploy-github-integration",
```
Comment (lzdanski) [inline sidebarsAstro.js:91]: Fix this and add to release notes

File: astro/deploy-github-integration.md [hunk 7]
```diff
@@ -72,7 +72,7 @@ Before you begin, ensure that you’re logged in to GitHub with permissions to r
 6. Configure the following fields:
 
     - **Repository:** Select the repository you want to integrate with Astro.
-    - **Astro Project Path:** Specify the path to your Astro project, up to and including the Astro project folder. For example, `/myorg/myprojects/my-astro-project`
+    - **Astro Project Path:** Specify the path to your Astro project relative to your GitHub repository directory, up to and including the Astro project folder. For example, for a project in `github-repository/myorg/myprojects/my-astro-project`, enter `myorg/myprojects/my-astro-project`. If the path is configured incorrectly, code does not deploy to Astro from GitHub. You can change this path by editing the **Branch Configuration** in the Astro UI.
```
Comment (jwitz) [inline astro/deploy-github-integration.md:75]:
```suggestion
    - **Astro Project Path:** Specify the path to your Astro project relative to your GitHub repository directory, up to and including the Astro project folder. For example, for a project in `github-repository/myorg/myprojects/my-astro-project`, enter `myorg/myprojects/my-astro-project`. If the path is configured incorrectly, code does not deploy to Astro from GitHub. You can change this path after you create the integration by editing the **Branch Configuration** setting in the Astro UI.
```

File: astro/release-notes.md [hunk 8]
```diff
@@ -38,6 +38,17 @@ Astronomer is committed to continuous delivery of both features and bug fixes to
 
 ## May 8, 2024
```
Comment (jwitz) [inline astro/release-notes.md:39]: Assume this date needs to change? Or better yet, might want to save this release note for the release notes PR

**Review by jwitz — APPROVED** [review]

File: astro/best-practices/connections-branch-deploys.md [hunk 9]
```diff
@@ -38,7 +38,7 @@ To implement this use case:
...
 2. Set these connections to be available to all Deployments in the Workspace by default. See [Configure connection sharing for a Workspace](create-and-link-connections.md#configure-connection-sharing-for-a-workspace). Your DAG authors can now access external development and testing resources across all Deployments.
 3. Choose a Deployment in your Workspace that you want to run your production workflows. In the Deployment, override the connections you configured to instead connect to your production resources. See [Override connection fields](create-and-link-connections.md#override-connection-fields).
-4. In the Git repository for your Astro project, define a multi-branch CI/CD pipeline for deploying to Astro. For an example of how to do this in GitHub, see the multibranch option in [CI/CD deploy action templates](/ci-cd-templates/github-actions-deploy-action.md#setup).
+4. In the Git repository for your Astro project, define a multi-branch CI/CD pipeline for deploying to Astro. For an example of how to do this in GitHub, see the [Astro Github integration](deploy-github-integration.md).
```
[the remark this answers is not in the document]
Comment (lzdanski) [reply astro/best-practices/connections-branch-deploys.md:41]: removed `natively` - style guide stuff https://developers.google.com/style/word-list#native

File: astro/ci-cd-templates/github-actions-deployment-preview.md [hunk 10]
```diff
@@ -8,6 +8,12 @@ description: Use pre-built Astronomer CI/CD templates to automate deploying Apac
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
+:::tip
+
+The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action, allowing you to see Git metadata directly from your Astro Deployment. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```
Comment (lzdanski) [inline astro/ci-cd-templates/github-actions-deployment-preview.md:13]:
```suggestion
The Astro GitHub integration can automatically deploy code from a GitHub repository to Astro without you needing to configure a GitHub action. In addition, the Astro UI shows Git metadata for each deploy on your Deployment information screen. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for setup steps.
```

File: astro/ci-cd-templates/github-actions-private-network.md [hunk 11]
```diff
@@ -8,7 +8,13 @@ description: Use pre-built Astronomer CI/CD templates to automate deploying Apac
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-If you don't have access to the Astronomer [deploy action](https://github.com/astronomer/deploy-action) because you can't access the public internet from your GitHub repository, use one of the following private network templates to deploy to Astro. 
+:::tip
+
+The Astro GitHub integration is a new way to automatically deploy code from a GitHub repository to Astro without needing to configure a GitHub Action, allowing you to see Git metadata directly from your Astro Deployment. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for more information.
```
Comment (lzdanski) [inline astro/ci-cd-templates/github-actions-private-network.md:13]:
```suggestion
The Astro GitHub integration can automatically deploy code from a GitHub repository to Astro without you needing to configure a GitHub action. In addition, the Astro UI shows Git metadata for each deploy on your Deployment information screen. See [Deploy code with the Astro GitHub integration](deploy-github-integration.md) for setup steps.
```

**Pull request merged**